src/renderer/components/header/Header.tsx

```diff
@@ -22,6 +26,13 @@ export const Header: React.FC = (): JSX.Element => {
     pools: { poolsState$, setSelectedPricePoolAsset: setSelectedPricePool, selectedPricePoolAsset$ },
     apiEndpoint$
   } = midgardService
+
+  const { network$, changeNetwork } = useAppContext()
+
+  const { removeAllLedgerAddress } = useChainContext()
```

Comment:
Can you try to handle `removeAllLedgerAddress` at service layer (by subscribing to `network$` or so - similar to resetting balances in your previous PR)?